telemetry-setup/action.yml

```diff
@@ -19,46 +19,15 @@ outputs:
 runs:
   using: composite
   steps:
+    - uses: actions/checkout@v4
```

Comment:
Based on the example below, you might be able to skip the `actions/checkout` step entirely by using the `GITHUB_ACTION_PATH` variable:

- https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action#creating-an-action-metadata-file

Those docs link to this page: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context, which has the snippet shown below:

![image](https://github.com/user-attachments/assets/c146da37-79ac-443e-a209-ed62c28d93ea)